lesson17/main.js

```diff
@@ -0,0 +1,74 @@
+const jsonUrl = "https://jsondata.okiba.me/v1/json/YbIZe211030061143";
+const wrapper = document.getElementById("js-wrapper");
+const ul = document.getElementById("js-slideshow");
+const numOfSecond = 3;
+
+const createElementWithClassName = (element, name) => {
+  const createdElement = document.createElement(element);
+  createdElement.classList.add(name);
+  return createdElement;
+};
+
+const createLoader = () => {
+  const loader = document.createElement("div");
+  const loaderImage = document.createElement("img");
+  loader.id = "js-loader";
+  loaderImage.src = "img/loading-circle.gif";
+  wrapper.appendChild(loader).appendChild(loaderImage);
+};
+
+const loading = () => {
+  const loader = document.getElementById("js-loader");
+  loader.classList.add("loading");
+};
+
+const loaded = () => {
+  const loader = document.getElementById("js-loader");
+  loader.classList.add("loaded");
+};
+
+const fetcheDataInSecond = (sec) => {
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      resolve(fetch(jsonUrl));
+    }, sec * 1000);
+  });
+};
+
+
+const fetcheImgData = async () => {
+  try {
+    const response = await fetcheDataInSecond(numOfSecond);
+    const json = await response.json();
+    return json.images;
+  } catch (error) {
+    wrapper.textContent = "データの取得ができませんでした。";
+    console.error(error);
+  } finally {
+    loaded();
+  }
+};
+
+const init = async () => {
+  createLoader();
+  loading();
+  const imgData = await fetcheImgData();
+  createListsOfImg(imgData);
+};
+
+const createListsOfImg = (imgData) => {
+  const fragment = document.createDocumentFragment();
+  imgData.forEach((imgVal,index) => {
+    const li = createElementWithClassName("li","slideshow-img");
+    const img = document.createElement("img");
+    img.src = imgVal.src;
+
+    index === 0 && li.classList.add("is-show");
+
```

Comment:
[ask]
I learned how to write this code for the first time! What is this being processed?

Reply:
This is the [logical and(&&)](https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Operators/Logical_AND)

In this situation, if the left side `index === 0` is true then the right side `li.classList.add("is-show")` will be executed. If left side is false, it just returns the left side.
It works the same way as below.
```
if (index === 0) {
      li.classList.add("is-show");
    }
```
I think the logical and is easier and more simple way to write than if statement, so I use it here :)
(I also got this advice from other student before!)